File: src/main/java/org/example/kalkulationsprogramm/domain/OutOfOfficeSchedule.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@Entity
@Table(name = "out_of_office_schedule")
public class OutOfOfficeSchedule {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(name = "start_at", nullable = false)
    private LocalDate startAt;

    @Column(name = "end_at", nullable = false)
    private LocalDate endAt;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "signature_id")
    private EmailSignature signature;

    @Column(name = "active", nullable = false)
    private boolean active;

    @Column(name = "subject_template", length = 300)
    private String subjectTemplate;

    @Lob
    @Column(name = "body_template", columnDefinition = "longtext")
    private String bodyTemplate;

}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/OutOfOfficeSchedule.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "out_of_office_schedule")
open class OutOfOfficeSchedule {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @Column(nullable = false, length = 200)
    open var title: String? = null

    @Column(name = "start_at", nullable = false)
    open var startAt: LocalDate? = null

    @Column(name = "end_at", nullable = false)
    open var endAt: LocalDate? = null

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "signature_id")
    open var signature: EmailSignature? = null

    @Column(name = "active", nullable = false)
    open var active: Boolean = false

    @Column(name = "subject_template", length = 300)
    open var subjectTemplate: String? = null

    @Lob
    @Column(name = "body_template", columnDefinition = "longtext")
    open var bodyTemplate: String? = null

    open fun isActive(): Boolean = active

}
